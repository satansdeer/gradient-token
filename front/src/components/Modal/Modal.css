.Modal-backdrop {
  background-color: rgba(0,0,0,0.3);
  height: 100%;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 100;
  opacity: 0;
  animation: backdropIn .25s cubic-bezier(.06,.67,.37,.99) forwards
}

.Modal {
  background-color: #fff;
  border-radius: 8px;
  max-width: 300px;
  padding: 20px;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  animation: modalIn .25s cubic-bezier(.06,.67,.37,.99) forwards
}

.Modal-close_button {
  background-color: rgb(251, 163, 113);
  border-radius: 4px;
  border: 2px solid #0b0b0b;
  box-shadow: rgb(255, 132, 75) 0px 0px 0px 3px inset;
  color: #fff;
  cursor: pointer;
  font-size: 16px;
  padding: 15px 20px;
}

@keyframes backdropIn {
  0%   { opacity: 0; }
  100% { opacity: 1; }
}

@keyframes modalIn {
  0%   { transform: translate(-50%, -50%) translateY(10px) }
  100% { transform: translate(-50%, -50%) }
}
