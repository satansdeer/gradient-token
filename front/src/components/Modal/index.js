import React from "react";
import { inject, observer } from "mobx-react";
import PropTypes from "prop-types";
import Button from "components/Button";
import "./Modal.css";

const Modal = ({ modalStore: { modal, hideModal } }) => {
  return modal.show ? (
    <div className={`Modal-backdrop ${modal.show && "is-visible"}`}>
      <div className="Modal">
        {modal.body}
        <Button onClick={hideModal} label="Close" />
      </div>
    </div>
  ) : null;
};

Modal.propTypes = {
  modalStore: PropTypes.shape({
    modal: PropTypes.shape({
      show: PropTypes.bool,
      body: PropTypes.object
    }),
    hideModal: PropTypes.func.isRequired
  }).isRequired
};

export default inject("modalStore")(observer(Modal));
