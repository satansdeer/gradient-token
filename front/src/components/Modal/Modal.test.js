import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Modal from "./index";

describe("<Modal />", () => {
  describe("when show is true and body provided", () => {
    const mockModalStore = {
      modal: { show: true, body: <div>Test</div> },
      hideModal: jest.fn()
    };
    const component = shallow(
      <Modal.wrappedComponent modalStore={mockModalStore} />
    );

    it("renders and matches the snapshot", () => {
      const component = renderer.create(
        <Modal.wrappedComponent modalStore={mockModalStore} />
      );
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it("calls hideModal method when close button is clicked", () => {
      component.find("Button").simulate("click");
      expect(mockModalStore.hideModal).toHaveBeenCalled();
    });
  });

  describe("when show is false and body is null", () => {
    const mockModalStore = {
      modal: { show: false, body: null },
      hideModal: jest.fn()
    };
    const component = shallow(
      <Modal.wrappedComponent modalStore={mockModalStore} />
    );

    it("returns null", () => {
      expect(component.getElement()).toBe(null);
    });
  });
});
