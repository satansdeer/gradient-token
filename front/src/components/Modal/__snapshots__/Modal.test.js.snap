// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> when show is true and body provided renders and matches the snapshot 1`] = `
<div
  className="Modal-backdrop is-visible"
>
  <div
    className="Modal"
  >
    <div>
      Test
    </div>
    <button
      className="Button undefined"
      onClick={[Function]}
    >
      Close
    </button>
  </div>
</div>
`;
